app/views/catalog/_bibliography_buttons_default.html.erb

```diff
@@ -0,0 +1,12 @@
+<% if @document.reference? %>
+  <% begin %>
+    <% url = URI.parse(@document.bibliography_url) %>
```

Comment:
This is great! Have you considered pulling this logic out of the view into a class? Perhaps that maybe useful as an extension point for other ways a BibTeX key could be expressed in a view?